superset/utils.py

```diff
@@ -572,3 +574,46 @@ def setup_cache(app, cache_config):
     """Setup the flask-cache on a flask app"""
     if cache_config and cache_config.get('CACHE_TYPE') != 'null':
         return Cache(app, config=cache_config)
+
+
+
+def zlib_compress(data):
+    """
+    compress things in a py2/3 safe fashion
+
+    >>> json_str = '{"test": 1}'
+    >>> blob = zlib_compress(json_str)
+    """
+
+    if PY3K:
+        if isinstance(data, str):
+            return zlib.compress(bytes(data, "utf-8"))
+        else:
```

Comment:
We may put a return zlib.compress(data) outside the if PY3K instead of the following 4 lines.